### PR TITLE
Do not install C sources in binary distributions

### LINKED ISSUE #6399: Binary distributions such as wheels include C sources
### Describe the bug

Binary distributions such as wheels include C sources (`_find_header.c`, `_find_header.h`, `_helpers.c`, `_http_parser.c`, `_http_writer.c`, and `_websocket.c`), but these sources don’t provide value in binary distributions and their size is significant.

See also https://github.com/aio-libs/frozenlist/issues/250.

### To Reproduce

```
$ python3 -m venv _e
$ . _e/bin/activate
(_e) $ pip install aiohttp
(_e) $ ls -l _e/lib64/python*/site-packages/aiohttp/*.{c,h,pyx,px*}
```

### Expected behavior

The C sources should ideally be present in the sdist but not in the bdist, where they don’t really provide any value.

### Logs/tracebacks

```python-traceback
$ python3 -m venv _e
$ . _e/bin/activate
(_e) $ pip install aiohttp
(_e) $ ls -l _e/lib64/python*/site-packages/aiohttp/*.{c,h,pyx,px*}
-rw-rw-r--. 1 ben ben   4998 Dec  7 23:42 _e/lib64/python3.10/site-packages/aiohttp/_cparser.pxd
-rw-rw-r--. 1 ben ben 187570 Dec  7 23:42 _e/lib64/python3.10/site-packages/aiohttp/_find_header.c
-rw-rw-r--. 1 ben ben    170 Dec  7 23:42 _e/lib64/python3.10/site-packages/aiohttp/_find_header.h
-rw-rw-r--. 1 ben ben     68 Dec  7 23:42 _e/lib64/python3.10/site-packages/aiohttp/_find_header.pxd
-rw-rw-r--. 1 ben ben   2007 Dec  7 23:42 _e/lib64/python3.10/site-packages/aiohttp/_headers.pxi
-rw-rw-r--. 1 ben ben 212982 Dec  7 23:42 _e/lib64/python3.10/site-packages/aiohttp/_helpers.c
-rw-rw-r--. 1 ben ben   1049 Dec  7 23:42 _e/lib64/python3.10/site-packages/aiohttp/_helpers.pyx
-rw-rw-r--. 1 ben ben 978187 Dec  7 23:42 _e/lib64/python3.10/site-packages/aiohttp/_http_parser.c
-rw-rw-r--. 1 ben ben  26571 Dec  7 23:42 _e/lib64/python3.10/site-packages/aiohttp/_http_parser.pyx
-rw-rw-r--. 1 ben ben 221270 Dec  7 23:42 _e/lib64/python3.10/site-packages/aiohttp/_http_writer.c
-rw-rw-r--. 1 ben ben   4575 Dec  7 23:42 _e/lib64/python3.10/site-packages/aiohttp/_http_writer.pyx
-rw-rw-r--. 1 ben ben 138420 Dec  7 23:42 _e/lib64/python3.10/site-packages/aiohttp/_websocket.c
-rw-rw-r--. 1 ben ben   1561 Dec  7 23:42 _e/lib64/python3.10/site-packages/aiohttp/_websocket.pyx
```

### Python Version

```console
$ python --version
Python 3.10.0
```

### aiohttp Version

```console
$ python -m pip show aiohttp
Name: aiohttp
Version: 3.8.1
[…]
```

### multidict Version

```console
$ python -m pip show multidict
Name: multidict
Version: 5.2.0
[…]
```

### yarl Version

```console
$ python -m pip show yarl
Name: yarl
Version: 1.7.2
[…]
```

### OS

Fedora Linux 35—but this issue is platform-independent

### Related component

Server

### Additional context

A suggested fix is to add to `setup(…)` in `setup.py`:

```python
    exclude_package_data={"": ["*.c", "*.h"]},
```

which will exclude these sources from binary distributions without affecting source distributions.

It’s possible to exclude some of the Cython files as well, but in `frozenlist` the preference was to keep them in the bdist for debugging purposes.

PR to follow.

### Code of Conduct

- [X] I agree to follow the aio-libs Code of Conduct

### PR DESCRIPTION
**Pull request by musicinmybrain**
This does not affect source distributions, and Cython sources (`.pyx`) are
still installed.

## What do these changes do?

Prevent C sources (`_find_header.c`, `_find_header.h`, `_helpers.c`, `_http_parser.c`, `_http_writer.c`, and `_websocket.c`) from being included in binary distributions such as wheels, without affecting Cython sources (`_cparser.pxd`, `_find_header.pxd`, `_headers.pxi`, `_helpers.pyx`, `_http_parser.pyx`, `_http_writer.pyx`, and `_websocket.pyx`) or source distributions.

## Are there changes in behavior for the user?

Nothing changes except that the binary wheels are much smaller because they do not contain ~1.7MB of C sources.

## Related issue number

Resolves #6399.

## Checklist

- [x] I think the code is well written
- [ ] Unit tests for the changes exist **This doesn’t seem necessary.**
- [ ] Documentation reflects the changes **No changes seem necessary.**
- [ ] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names.
  * **This contribution is trivial, so this doesn’t seem necessary, but I can add it if  desired.**
- [x] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -71,9 +71,13 @@ exclude =
 # https://setuptools.readthedocs.io/en/latest/setuptools.html#options
 # (see notes for the asterisk/`*` meaning)
 * =
-#    *.c
     *.so
 
+[options.exclude_package_data]
```
Comment (webknjaz) [inline setup.cfg]: plz move this before `extra_requires` to keep the sections sorted alphabetically.

File: CHANGES/6399.bugfix [hunk 2]
```diff
@@ -0,0 +1 @@
+Do not install C sources with binary distributions.
```
Comment (webknjaz) [inline CHANGES/6399.bugfix]: I don't think it's a `bugfix`. Maybe it's a `misc` but definitely not something affecting the end-users' runtime.

**Review by webknjaz — COMMENTED** [review]
Not sure how I feel about this in general but here's some actionable feedback for now.

**Comment by webknjaz** [comment]
> Not sure how I feel about this in general but here's some actionable feedback for now.

Based on Andrew's comments in the issue, this will be good to go once those comments are addressed.

**Review by webknjaz — APPROVED** [review]

**Comment by musicinmybrain** [comment]
https://github.com/aio-libs/aiohttp/runs/4458544871?check_suite_focus=true#step:10:2941

> E               ResourceWarning: unclosed transport <_SelectorSocketTransport closing fd=46>

Doesn’t look like the CI failure has anything to do with the PR.

**Comment by webknjaz** [comment]
> Doesn’t look like the CI failure has anything to do with the PR.

That's probably a race condition in tests. It's not related to the CI itself but the tests could be improved, for sure. And of course, it's a separate effort/PR. I'm restarting the CI for now.

**Pull request merged**